services/src/main/java/io/grpc/services/BinaryLogProviderImpl.java

```diff
@@ -34,12 +39,25 @@
   private final AtomicLong counter = new AtomicLong();
 
   public BinaryLogProviderImpl() {
-    this(BinaryLogSinkProvider.provider(), System.getenv("GRPC_BINARY_LOG_CONFIG"));
+    this(new TempFileSink(), System.getenv("GRPC_BINARY_LOG_CONFIG"));
   }
 
-  BinaryLogProviderImpl(BinaryLogSink sink, String configStr) {
+  BinaryLogProviderImpl(final BinaryLogSink sink, String configStr) {
     BinaryLog.Factory factory = null;
     try {
+      Preconditions.checkNotNull(sink);
+      Runtime.getRuntime().addShutdownHook(new Thread() {
```

Comment:
We can't be doing this in a library.

We can't even assume our library should be loaded when the JVM exits. For example, in servlet environments we could be unloaded very frequently, but this would retain our class loader.